friprosveta/views.py

```diff
@@ -478,6 +479,46 @@ def allocations_json(request, timetable_slug=None):
     filtered_allocations = _allocation_set(param_ids,
                                            tt.allocations,
                                            request.user.is_staff)
+
+    #returns more information about allocations
+    r = request.GET
+    if 'mode' in r:
+        l = r['mode']
+        if l == 'ext':
+            AllocationVM = namedtuple('AllocationVM', ['object', 'day_index', 'hour_index', 'duration'])
+            weekday_mapping = {wd[0]: i for i, wd in enumerate(WEEKDAYS)}
+            hour_mapping = {wh[0]: i for i, wh in enumerate(WORKHOURS)}
+            allocation_vms = [AllocationVM(
+                object=a,
+                day_index=weekday_mapping[a.day],
+                hour_index=hour_mapping[a.start],
+                duration=a.duration,
+            ) for a in filtered_allocations]
+
+            allocation_vms = sorted(allocation_vms, key=lambda avm: avm.day_index)
+            allocations_by_day = [(d, list(avm_grouper))
+                                  for d, avm_grouper in itertools.groupby(allocation_vms, lambda avm: avm.object.day)]
+            allocations_ext = dict() 
+            for a in allocations_by_day:
+                allocations_day = []
+                for predmet in a[1]:
+                    prObj =  predmet[0]
```

Comment:
Is predmet not a namedtuple? If so, please access properties by names (so `object` or even better `allocation` instead ou `[0]`).

Reply:
Code corrected.